modules/vaos/app/services/eps/appointment_service.rb

```diff
@@ -31,9 +31,29 @@ def get_appointments
     ##
     # Create draft appointment in EPS
     #
+    # @param referral_id [String] The ID of the referral to create an appointment for
+    # @raise [ArgumentError] If referral_id is blank
+    # @raise [Common::Exceptions::ValidationErrors] If an appointment already exists for the referral
     # @return OpenStruct response from EPS create draft appointment endpoint
     #
     def create_draft_appointment(referral_id:)
+      raise ArgumentError, 'referral_id is required and cannot be blank' if referral_id.blank?
+
+      # Get all appointments and check for existing ones with this referral ID
+      existing_appointments = get_appointments
+      if existing_appointments.respond_to?(:each)
+        existing_appointment = existing_appointments.find do |appt|
+          appt.referral_id == referral_id &&
```

Comment:
Note: FYI - EPS returns a referral_number which is then assigned to referral_id in its model.